### PR TITLE
Improve platform views performance

### PR DESCRIPTION
**Pull request by blasten**
Improves the performance of platform views by rendering to a TextureLayer
while a view is embedded to the Android view hierarchy.

This is possible because a `View` can be drawn to an arbitrary `Canvas`.
For example:

```java
SurfaceTexture tx = new SurfaceTexture(0);
Surface surface = new Surface(tx);
FrameLayout view = new FrameLayout(context) {
    @Override
    public void draw(Canvas canvas) {
        final Canvas newCanvas = surface.lockHardwareCanvas();
        try {
            super.draw(newCanvas);
        } finally {
            surface.unlockCanvasAndPost(newCanvas);
        }
    }
}
// use texture in the GPU thread.
```

`Surface#lockHardwareCanvas` is available since API level 23.

This PR eliminates the Virtual Display backend implementation, and replaces it
for hybrid composition rendered as a texture layer. 

This change should be transparent to clients that rely on Virtual Display implementations.

### Next steps

In the current hybrid composition implementation, this change removes the need for 
raster synchronization (e.g. thread merging and FlutterImageView).

### Issue

https://github.com/flutter/flutter/issues/96679

### CODE REVIEW

**Review by jason-simmons — COMMENTED** [review]
LGTM

File: shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureWrapper.java [hunk 1]
```diff
@@ -62,7 +62,11 @@ public void attachToGLContext(int texName) {
   // Called by native.
   @SuppressWarnings("unused")
   public void detachFromGLContext() {
-    surfaceTexture.detachFromGLContext();
+    synchronized (this) {
+      if (!released) {
```
[the remark this answers is not in the document]
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/engine/renderer/SurfaceTextureWrapper.java]: fixed in https://github.com/flutter/engine/pull/31390

**Review by cyanglaz — APPROVED** [review]
LGTM on platform view part. I only have some general ideas of how android APIs used in this PR work; I think we should add someone who is more familiar with android to take another look at the android code.

File: shell/platform/android/io/flutter/embedding/android/FlutterView.java [hunk 2]
```diff
@@ -867,21 +866,6 @@ public InputConnection onCreateInputConnection(@NonNull EditorInfo outAttrs) {
...
-  /**
-   * Allows a {@code View} that is not currently the input connection target to invoke commands on
-   * the {@link android.view.inputmethod.InputMethodManager}, which is otherwise disallowed.
-   *
-   * <p>Returns true to allow non-input-connection-targets to invoke methods on {@code
-   * InputMethodManager}, or false to exclusively allow the input connection target to invoke such
-   * methods.
-   */
-  @Override
-  public boolean checkInputConnectionProxy(View view) {
```
[the remark this answers is not in the document]
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterView.java:879]: this was only used in virtual displays. There's no virtual displays now, so there's no need for this method.

File: shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java [hunk 3]
```diff
@@ -82,29 +86,40 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result
         }
 
         private void create(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
-          Map<String, Object> createArgs = call.arguments();
-          boolean usesHybridComposition =
+          final Map<String, Object> createArgs = call.arguments();
+          // TODO(egarciad): Remove the "hybrid" case.
```
[the remark this answers is not in the document]
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java:90]: done

File: shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java [hunk 4]
```diff
@@ -249,36 +272,38 @@ public void setPlatformViewsHandler(@Nullable PlatformViewsHandler handler) {
...
+     * The Flutter application would like to resize an existing Android {@code View}.
+     *
+     * @param request The request to resize the platform view.
+     * @return The buffer size where the platform view pixels are written to.
      */
-    void resizePlatformView(
-        @NonNull PlatformViewResizeRequest request, @NonNull Runnable onComplete);
+    PlatformViewBufferSize resize(@NonNull PlatformViewResizeRequest request);
+
+    /** The Flutter application would like to change the offset an existing Android {@code View}. */
```
[the remark this answers is not in the document]
Comment (blasten) [reply shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformViewsChannel.java]: done

**Pull request merged**